lib/hooks.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import * as TrackPlayer from './index';
+import TrackPlayerEvents from './eventTypes';
+
+function isValidTrackPlayerEvent(evt) {
+    return TrackPlayerEvents[evt] !== undefined;
+}
+
+/**
+ * @description
+ *   Attaches handler to TrackPlayer event and performs cleanup on unmount
+ */
+module.exports.useTrackPlayerEvent = (event, handler) => {
```

Comment:
would it make sense for this to be able to take an array of event types? so that you could use this callback, to subscribe to multiple? instead of having a few callbacks?

Reply:
Also, why not move this function to `index.js`?

Reply:
I can make this accept an array of event types :+1:. Should we also log a dev-only warning when the library user has a typo or invalid event type? (i.e., `TrackPlayerEventTypes[event] !== undefined`). 

I opted to move this into its own file because it seemed like `index.js` was already sizable, but I can move it into `index.js` if you want. 

I am also not sure if users under RN v0.59 will see errors if this is part of `index.js` when they do `import * as TrackPlayer from 'react-native-track-player'` in their code since `useEffect` will not exist in React for them (yet). It might only error when they attempt to use the function.

I also noticed there's not really a JS build step in the repo. I'd be happy to set one up for the project as well (in a separate PR). This would fix any potential import problems that a library consumer might have if they aren't compiling the ES6+ plus the added benefit of a minified lib to be used (plus allows anyone to use esnext features in this repo).